### PR TITLE
add 'title' field for custom actions

### LINKED ISSUE #242: Support display names for actions
The CNAB WD says that action names "SHOULD use dotted name syntax".  This is great for uniqueness, but not great for UI.  Could we have an optional 'display name' attribute for actions?  The scenario is a GUI which displays the actions that can be taken on a bundle (or installation/instance/claim), e.g. as buttons or on a menu: it would be nice to be able to display:

```
--------------------------------------
|  Upgrade  |  Status  |  Uninstall  |
--------------------------------------
```

instead of

```
----------------------------------------------
|  Upgrade  |  io.cnab.status  |  Uninstall  |
----------------------------------------------
```

### PR DESCRIPTION
**Pull request by technosophos**
Adds a `display` field for custom actions so that a human friendly name can be shown for each custom action.

Closes #242 

Signed-off-by: Matt Butcher <matt.butcher@microsoft.com>

### CODE REVIEW

File: 101-bundle-json.md [hunk 1]
```diff
@@ -743,14 +743,17 @@ Implementations MAY support user-defined additional actions as well. Such action
 {
   "actions": {
     "io.cnab.dry-run": {
+      "display": "Dry Run",
       "description": "prints what install would do with the given parameters values",
       "modifies": false,
       "stateless": true
     },
     "io.cnab.migrate": {
+      "display": "Migratte",
```
Comment (vdice) [inline 101-bundle-json.md]: I believe this should be `"Magritte"` 😉 

art-magritte-listening-chamber-450x367px

File: schema/bundle.schema.json [hunk 2]
```diff
@@ -187,6 +187,10 @@
             "description": "A description of the purpose of this action",
             "type": "string"
           },
+          "display": {
+            "description": "A human-readable name for thiis action",
```
Comment (vdice) [inline schema/bundle.schema.json]: Too many 👁 s in `thiis`

File: 101-bundle-json.md [hunk 3]
```diff
@@ -764,6 +767,7 @@ The above declares three actions: `io.cnab.status`, `io.cnab.migrate` and `io.cn
 
 Each action is accompanied by a description, which contains the following fields:
 
+- `display`: The human-readable name of the operation. If omitted, tools MAY use the value of the `actions` key (e.g. `io.cnab.dry-run`). User agents that display the display name SHOULD disambiguate if two or more actions have the same `display` value.
```
Comment (silvin-lubecki) [inline 101-bundle-json.md]: nit: We use `title` in the `definitions` specs, which comes from JSONSchema, see https://github.com/cnabio/cnab-spec/blob/36d7b4817de6902142fc338549c8d93aa94b37a5/101-bundle-json.md#L411

**Comment by technosophos** [comment]
I have updated this to use `title` instead of `display`. I agree that sounds much better.

File: schema/bundle.schema.json [hunk 4]
```diff
@@ -187,6 +187,10 @@
             "description": "A description of the purpose of this action",
             "type": "string"
           },
+          "display": {
```
Comment (vdice) [inline schema/bundle.schema.json]: This should change to `title`, correct?

File: 101-bundle-json.md [hunk 5]
```diff
@@ -764,6 +767,7 @@ The above declares three actions: `io.cnab.status`, `io.cnab.migrate` and `io.cn
 
 Each action is accompanied by a description, which contains the following fields:
 
+- `title`: The human-readable name of the operation. If omitted, tools MAY use the value of the `actions` key (e.g. `io.cnab.dry-run`). User agents that display the title SHOULD disambiguate if two or more actions have the same `title` value.
```
Comment (vdice) [inline 101-bundle-json.md]: Should we add `(OPTIONAL)` at the end of this entry, as is done elsewhere?  (Technically it looks like strict adherence isn't already in place as `modifies` below also doesn't have this...)

File: schema/bundle.schema.json (hunk 4 above)
Comment (technosophos) [reply schema/bundle.schema.json]: Ugh... I missed a file in my commit. Sorry. Will fix that.

File: 101-bundle-json.md (hunk 5 above)
Comment (technosophos) [reply 101-bundle-json.md]: Yes. Good catch.

**Pull request merged**